charts/opentelemetry-collector/values.yaml

```diff
@@ -40,6 +40,20 @@ presets:
   # Best used with mode = daemonset.
   kubernetesAttributes:
     enabled: false
+  # Configures the collector to collect kubernetes objects.
+  # Adds the k8sobject receiver to the logs pipeline
+  # and collects kubernetes events by default.
+  # Best used with mode = deployment or statefulset.
+  kubernetesObjects:
+    enabled: false
+    defaultEventsConfig:
+      mode: 'watch'
+      labelSelector: ""
+      fieldSelector: ""
+      interval: 60m
+      resourceVersion: ""
+      namespaces: []
+      group: 'events.k8s.io'
```

Comment:
I'm not sure we should introduce another limited configuration interface layer for k8sobjects. We should allow users to introduce new rules, e.g., `watch` and `list`.

We can have a default the most "common" config, which users will be able to override with:

```yaml
config:
  recievers:
    k8sobjects:
      objects:
         ...
```

Reply:
@dmitryax @nicolastakashi since this is a preset I would opt for default, out-of-the-box content that can be enabled with `.Values.presets.kubernetesObjects.enabled` and keep the `kubernetesObjects` configuration very simple.  If there are extra scenarios we need to allow, we should try to handle it like we do for `logsCollection`'s scenarios.  Our goal with the preset quick, easy-to-use configuration that provides fast, basic value.

Any complex configuration falls outside the expectations for a preset; in those situations the component should be configured manually.

Are there meaningful values we can use as defaults for this component?

Reply:
IMH the values provided are meaningful and provide a default, out-of-the-box content, the extensibility we have under the `defaultEventsConfig` don't change or include any feature on the existing preset, only changes some basic configs like default mode if it will be watch or pull and so on.

Reply:
Ok I understand now how this default section interacts with the template in _config.tpl.  Since the user is able to add to/modify a preset via `.Values.config` I would vote we don't need a `defaultEventsConfig` section.  If  `label_selector`, `field_selector`, or `namespaces` are pretty important options for the component maybe it is worth it to include those next to `enable`, but we could hard code the rest.  @dmitryax @povilasv what do you think?

Reply:
Don't think they are very important if we don't specify we will have a default behavior, I'm ok with not having those fields available for overriding preset config.

But let's see what the guys think about 😄

Reply:
@povilasv  I like the suggestion. And calling preset `kubernetesEvents` makes more sense. Collection of k8s objects in pull mode seems like a far less common use case and can always be configured via `config` section.

I'd keep the preset slim:

```yaml
presets:
  kubernetesEvents:
    enabled: true
```

Reply:
Hey folks!
Another push, can you take a look @dmitryax ?